src/data/nat/basic.lean

```diff
@@ -1674,4 +1674,56 @@ lemma decreasing_induction_succ_left {P : ℕ → Sort*} (h : ∀n, P (n+1) →
 by { rw [subsingleton.elim mn (le_trans (le_succ m) smn), decreasing_induction_trans,
          decreasing_induction_succ'] }
 
+/-- `log hb n`, is the logarithm of natural number
+`n` in base `b`. If `n` is not a power of `b`, it returns the
+largest `k:ℕ` such that `b^k ≤ n`. -/
+def log (b : ℕ) : ℕ → ℕ
+| n :=
+  if h : b ≤ n ∧ 1 < b then
+    have n / b < n,
+      from div_lt_self
+        (nat.lt_of_lt_of_le (lt_trans zero_lt_one h.2) h.1) h.2,
+    log (n / b) + 1
+  else 0
+
+lemma exp_le_iff_le_log (x y : ℕ) {b} (hb : 1 < b) (hy : 1 ≤ y) :
+  b^x ≤ y ↔ x ≤ log b y :=
+begin
+  induction y using nat.strong_induction_on with y ih
+    generalizing x,
+  rw [log], split_ifs,
+  { have h'' : 0 < b := lt_of_le_of_lt (zero_le _) hb,
+    cases h with h₀ h₁,
+    rw [← nat.sub_le_right_iff_le_add,← ih (y / b),
+          le_div_iff_mul_le _ _ h'',← nat.pow_succ],
+    { cases x; simp [h₀,hy] },
+    { apply div_lt_self; assumption },
+    { rwa [le_div_iff_mul_le _ _ h'',one_mul], } },
+  { replace h := lt_of_not_ge (not_and'.1 h hb),
+    split; intros h',
+    { have := lt_of_le_of_lt h' h,
+      apply le_of_succ_le_succ,
+      change x < 1, rw [← pow_lt_iff_lt_right hb,pow_one],
+      exact this },
+    { replace h' := le_antisymm h' (zero_le _),
+      rw [h',nat.pow_zero], exact hy} },
+end
+
+lemma log_exp (b x : ℕ) (hb : 1 < b) : log b (b ^ x) = x :=
+eq_of_forall_le_iff $ λ z,
+by { rwa [← exp_le_iff_le_log _ _ hb,pow_le_iff_le_right],
+     rw ← nat.pow_zero b, apply pow_le_pow_of_le_right,
+     apply lt_of_le_of_lt (zero_le _) hb, apply zero_le }
+
+lemma exp_succ_log_gt_self (b x : ℕ) (hb : b > 1) (hy : 1 ≤ x) :
```

Comment:
Is the name of this consistent with the rest of mathlib? First of all, I think `exp` is currently only used for `e^x`. Also, is there a reason why the name has `gt` in it? This could probably be named `lt_pow_succ_log_self` or something similar.